types/react-redux/react-redux-tests.tsx

```diff
@@ -899,3 +899,25 @@ namespace TestCreateProvider {
     // <h1>A is 2</h1>
     ReactDOM.render(<Combined />, document.body);
 }
+
+namespace TestTypeInference {
```

Comment:
Good call on these tests here. In fact, perhaps we should add tests for all 12 overloads. That's outside the scope of this PR though, just thinking aloud :)